### PR TITLE
consensus: introduce faker and replace gxhash with faker

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes

- This PR introduces faker which is derived from gxhash. The consensus engine for tests are replaced to faker.
- In later PR, the gxhash and clique will be removed.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [ ] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: consensus/faker/faker_test.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+	header2 := &types.Header{Number: big.NewInt(5)}
+	err = f2.VerifySeal(nil, header2)
+	assert.Error(t, err)
+}
+
+// TestVerifyHeaders tests batch header verification
+func TestVerifyHeaders(t *testing.T) {
+	f := NewFaker()
+
+	// Test with empty headers
```
Comment (2dvorak) [inline consensus/faker/faker_test.go:174]: I cannot guess the point of this test, could you explain?

Comment (yoomee1313) [reply consensus/faker/faker_test.go:174]: This test is checking the edge case where the caller provides no headers at all (of course it's not happening). I wan't to make it sure it's not panic.

Comment (2dvorak) [reply consensus/faker/faker_test.go:174]: Got it, thanks.
